{
  exception=new NamedParamsException();
  assertNotNull(exception);
  assertEquals("\nnull\n",exception.getMessage());
  exception=new NamedParamsException("message");
  assertEquals("\nmessage\n",exception.getMessage());
  Throwable t=mock(Throwable.class);
  when(t.getStackTrace()).thenReturn(new StackTraceElement[0]);
  exception=new NamedParamsException(t);
  assertEquals(t,exception.getCause());
  exception=new NamedParamsException("message",t);
  assertEquals("\nmessage\n",exception.getMessage());
  assertEquals(t,exception.getCause());
}
